code/code/misc/periodic.cc

```diff
@@ -2080,8 +2080,10 @@ int TObj::updateBurning(void)
     // suppose we have a wooden staff, 4000 volume, 20 structure and 500
     // flammability, (20/4000)=.005, .005*500 = 2.5, 20/2.5 = 8
     // so it will burn for approx 8 rounds
-    int burnamount = (int)((double)((double)getMaxStructPoints()/getVolume())*
-			       material_nums[getMaterial()].flammability);
+
+    // (getVolume() | 1) avoids division by zero edge case
```

Comment:
Huh, what a "creative" approach.
is it OK that a volume of 100 now magically starts to be 101?
Why the double cast to `double`? Why the explicit cast to int?

Reply:
> Huh, what a "creative" approach.

Guessing that's sarcasm, haha. 

> is it OK that a volume of 100 now magically starts to be 101?

Not entirely sure what you mean here, which makes me think that I misunderstood how short circuiting works in C++. From looking it up I was under the impression that doing the `(getVolume() | 1)` there would behave similarly to how I use them in JavaScript, where if `getVolume()` resolves to a truthy value it will ignore the `| 1`, whereas if `getVolume()` is falsy it will replace it with the value on the other side of the `|` operator. But I take it that's not correct?

> Why the double cast to double? Why the explicit cast to int?

I don't honestly know - for a lot of this stuff I'm just trying to get the the code to work within the context of the original intent/methodology. Here I just saw that they were getting runtime errors for division by 0, so I tried to fix that specific part of the problem. Maybe I should've looked into it more deeply, but I'm already a little beyond my depth as you can probably tell.

Reply:
Nah, that'd be `||`. By doing `a | 1` you're setting the last bit to 1, no matter what the value of a is. So if a is 0 or 1, you end up with 1. For any even number, you get number+1.

Reply:
Regarding the casts - the usual way is to omit them, unless you actually need an explicit cast. Here, you might need just one cast - so that the division would be fractional, not whole-integer division. The rest look redundant, to me.

Regarding "within the context" - well, the code might have been good, once upon a time, but it's gotten a bit dusty and rusty. Best if we clean up anything we touch.